aidial_assistant/model/model_client.py

```diff
@@ -1,22 +1,34 @@
 from abc import ABC
-from typing import Any, AsyncIterator, List, TypedDict
+from typing import Any, AsyncIterator, List
 
-import openai
 from aidial_sdk.chat_completion import Role
-from aiohttp import ClientSession
+from aidial_sdk.utils.merge_chunks import merge
+from openai import AsyncOpenAI
 from pydantic import BaseModel
 
+from aidial_assistant.utils.open_ai import ToolCall, Usage
+
 
 class ReasonLengthException(Exception):
     pass
 
 
 class Message(BaseModel):
     role: Role
-    content: str
+    content: str | None = None
+    tool_call_id: str | None = None
+    tool_calls: list[ToolCall] | None = None
 
     def to_openai_message(self) -> dict[str, str]:
-        return {"role": self.role.value, "content": self.content}
+        result = {"role": self.role.value, "content": self.content}
```

Comment:
Why not `return self.dict()` or `return self.dict(exclude_none=True)` ?

Reply:
Reusing the message class from openai client.